misc/python/materialize/cli/mkrelease.py

```diff
@@ -274,6 +276,67 @@ def update_versions_list(released_version: Version) -> None:
         raise errors.MzRuntimeError("Couldn't determine where to insert new version")
 
 
+def update_upgrade_tests(released_version: Version) -> None:
+    upgrade_dir = Path("./test/upgrade")
+    version = f"v{released_version}"
+    need_upgrade = [
+        str(p) for p in upgrade_dir.glob("*latest_version*") if "smoke" not in str(p)
+    ]
+    if not need_upgrade:
+        return
+    for path in need_upgrade:
+        spawn.runv(["git", "mv", path, path.replace("latest_version", version)])
+
+    mzcompose = upgrade_dir.joinpath("mzcompose.yml")
+    with mzcompose.open() as fh:
+        contents = fh.readlines()
+
+    workflow_version = str(released_version).replace(".", "_")
+    step = f"""
+      - step: workflow
+        workflow: upgrade-from-{workflow_version}
+"""
+    for i, line in enumerate(contents):
+        if "workflow: upgrade-from-latest" in line:
+            contents.insert(i - 2, step)
+            break
+
+    previous = None
+    tests = None
+    insert_workflow = 0
+    for i, line in enumerate(contents):
+        if "PREVIOUS_VERSION:" in line:
+            previous = line.split(":")[1].strip()
+        if "TESTS:" in line:
+            tests = line.split(":")[1].strip()
+
+        if "upgrade-from-latest:" in line:
+            insert_workflow = i - 1
+            workflow = f"""
+  upgrade-from-{workflow_version}:
+    env:
+      PREVIOUS_VERSION: {previous}
+      TESTS: any_version|{tests}|{version}
+    steps:
+      - step: workflow
+        workflow: test-upgrade
+"""
+
+        if insert_workflow and "TESTS:" in line:
```

Comment:
The splicing of the new section in mzcompose.yml appears to be incorrect:

```
  upgrade-from-latest:
    env:

  upgrade-from-0_8_4:
    env:
      PREVIOUS_VERSION: v0.8.3
      TESTS: any_version|any_version|v0.6.1|v0.7.3|v0.8.0|v0.8.1|v0.8.2|v0.8.3|v0.8.4
    steps:
      - step: workflow
        workflow: test-upgrade
      PREVIOUS_VERSION: latest
      TESTS: any_version|v0.6.1|v0.7.3|v0.8.0|v0.8.1|v0.8.2|v0.8.3|v0.8.4|latest_version
    steps:
      - step: workflow
        workflow: test-upgrade
```

So the  ```upgrade-from-0_8_4``` section landed in the middle of the ```upgrade-from-latest``` section.

To make life easier, you may wish to:
1. rearrange mzcompose.yml so that all the ``upgrade-from``` sections are at the very bottom
2. put a comment at the very end of the file that the items below that comment are generated automatically by mkrelease.py
3. Have mkrelease.py append the new section at the very end of the file rather than trying to splice it in.

Reply:
That's fair. I've now modified the script to correctly use the the "insert_workflow" variable that should be more robust to slight changes.